tools/nightly_build_and_test_stats.groovy

```diff
@@ -0,0 +1,107 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import groovy.json.JsonSlurper
+
+node ("master") {
+  def jenkinsUrl = "${params.JENKINS_URL}"
+  def trssUrl    = "${params.TRSS_URL}"
+
+  def buildFailures = 0
+  def testStats = []
+
+  // Get the number of "Failed Builds"
+  stage("getBuildFailures") {
+    def builds = sh(returnStdout: true, script: "wget -q -O - ${jenkinsUrl}/view/Failed%20Builds/api/json")
+    def json = new JsonSlurper().parseText(builds)
+    buildFailures = json.jobs.size()
+  }
+
+  // Get the last Nightly test job & case stats
+  stage("getTestStats") {
+    // Get top level builds names
+    def trssBuildNames = sh(returnStdout: true, script: "wget -q -O - ${trssUrl}/api/getTopLevelBuildNames?type=Test")
+    def buildNamesJson = new JsonSlurper().parseText(trssBuildNames)
+    buildNamesJson.each { build ->
+      // Is it a build Pipeline?
+      if (build._id.buildName.contains("-pipeline")) {
+        echo "Pipeline ${build._id.buildName}"
+        def pipelineName = build._id.buildName
+        def pipelineUrl
+        def testJobSuccess = 0
+        def testJobUnstable = 0
+        def testJobFailure = 0
+        def testCasePassed = 0
+        def testCaseFailed = 0
+        def testCaseDisabled = 0
+
+        // Get all pipeline builds started by "timer", as those are Nightlies
+        def pipeline = sh(returnStdout: true, script: "wget -q -O - ${trssUrl}/api/getBuildHistory?buildName=${pipelineName}\\&startBy=timer")
+        def pipelineJson = new JsonSlurper().parseText(pipeline)
+        if (pipelineJson.size() > 0) {
+          // First in list is last Nightly job
+          def pipeline_id = pipelineJson[0]._id
+          pipelineUrl = pipelineJson[0].buildUrl 
+
+          // Get all child Test jobs for this pipeline job
+          def pipelineTestJobs = sh(returnStdout: true, script: "wget -q -O - ${trssUrl}/api/getAllChildBuilds?parentId=${pipeline_id}\\&buildNameRegex=^Test_.*")
+          def pipelineTestJobsJson = new JsonSlurper().parseText(pipelineTestJobs)
+          if (pipelineTestJobsJson.size() > 0) {
```

Comment:
Presumably the number of child jobs is not changing often.  It changes when we add more test targets into the build pipelines (which we will do soon, plan to add extended.openjdk).  When this happens, there will be a spike in failures (and disabled jobs).  

It may be worthwhile noting how many children there are, to explain one case where numbers might suddenly change.

Reply:
Added test job number